packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js

```diff
@@ -219,9 +219,9 @@ const CustomerForm = props => {
                 <div className={classes.region}>
                     <Region
                         validate={isRequired}
-                        fieldInput={'region[region]'}
-                        fieldSelect={'region[region_id]'}
-                        optionValueKey="id"
+                        fieldInput={'region[code]'}
+                        fieldSelect={'region[code]'}
+                        optionValueKey="code"
```

Comment:
`code` is the default prop for `optionValueKey` but I can't remember why we switched to `id` here in the first place. Seems like we also use `id` and `region[region]`/`region[region_id]` in [addEditDialog.js](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/AddressBookPage/addEditDialog.js#L138) as well.

Reply:
Good point but unfortunately [addEditDialog.js](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/AddressBookPage/addEditDialog.js#L138) needs to be changed as well. Everywhere in graphql the returned response has `code` as the unique key to identify the region. 

For instance here is a sample region object when the `optionValueKey="id"`
![image](https://user-images.githubusercontent.com/35203638/111359967-2fad5780-865a-11eb-9cf0-6e75332e7be9.png)

And here is the same object with `optionValueKey="code"`
![image](https://user-images.githubusercontent.com/35203638/111360024-405dcd80-865a-11eb-810a-adfbfc98dc08.png)

We have to use `code` instead of `id` for `optionValueKey` everywhere a region is used.

Shall I push the change for `addEditDialog.js` as part of this PR?

Reply:
Let's talk about this in dev time and update back here with a comment because I don't recall enough context to make this decision.